### PR TITLE
Adding some twitch functionality that is needed an documented

### PR DESCRIPTION
**Pull request by boostchicken**
Login Tag
Cumulative Months
Sub and Resub Msg-Ids

### CODE REVIEW

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/MsgId.java [hunk 1]
```diff
@@ -50,6 +50,16 @@
         private KnownValues() {
         }
 
+        /**
+         * Sub
+         */
+        public static final String SUB = "sub";
+
+        /**
+         * Resubed
```
Comment (lol768) [inline src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/MsgId.java]:
```suggestion
         * Resubbed
```

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    /**
+     * Function to create this message tag.
+     */
+    @SuppressWarnings("ConstantConditions")
+    public static final TriFunction<Client, String, String, Login> FUNCTION = (client, name, value) -> new Login(name, value);
+
+    private Login(@NonNull String name, @NonNull String value) {
+        super(name, value);
+    }
+}
```
Comment (lol768) [inline src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java]:
```suggestion
}
```

Comment (boostchicken) [reply src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java]: Confused here? No changes?

Comment (lol768) [reply src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java]: Missing `\n` :)

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/TwitchSupport.java [hunk 3]
```diff
@@ -26,31 +26,7 @@
...
-import org.kitteh.irc.client.library.feature.twitch.messagetag.RoomId;
-import org.kitteh.irc.client.library.feature.twitch.messagetag.Slow;
-import org.kitteh.irc.client.library.feature.twitch.messagetag.SubsOnly;
-import org.kitteh.irc.client.library.feature.twitch.messagetag.Subscriber;
-import org.kitteh.irc.client.library.feature.twitch.messagetag.SystemMsg;
-import org.kitteh.irc.client.library.feature.twitch.messagetag.Turbo;
-import org.kitteh.irc.client.library.feature.twitch.messagetag.User;
-import org.kitteh.irc.client.library.feature.twitch.messagetag.UserId;
-import org.kitteh.irc.client.library.feature.twitch.messagetag.UserType;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.*;
```
Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/feature/twitch/TwitchSupport.java]: Please revert this change - no `*` imports.

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package org.kitteh.irc.client.library.feature.twitch.messagetag;
```
Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java]: Missing license header.

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/MsgParamCumulativeMonths.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package org.kitteh.irc.client.library.feature.twitch.messagetag;
```
Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/MsgParamCumulativeMonths.java]: Missing license header.

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/TwitchSupport.java (hunk 3 above)
Comment (lol768) [reply src/main/java/org/kitteh/irc/client/library/feature/twitch/TwitchSupport.java]: We don't have a style guide..

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java (hunk 4 above)
Comment (mbax) [reply src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java]: This can be resolved by just running maven once, FYI :)

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/MsgParamCumulativeMonths.java (hunk 5 above)
Comment (mbax) [reply src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/MsgParamCumulativeMonths.java]: This can be resolved by just running maven once, FYI :)

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/TwitchSupport.java (hunk 3 above)
Comment (mbax) [reply src/main/java/org/kitteh/irc/client/library/feature/twitch/TwitchSupport.java]: True, but I'll get around to doing that darnit! And yeah, please revert the import change. Or I can do it before I pull it to save you time! :)

**Comment by mbax** [comment]
Thanks! Hoping to actually review this tomorrow. Just got back from a trip, and am pretty exhausted. Please don't consider my couple comments added to be my only thoughts on this :D

File: src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java (hunk 4 above)
Comment (boostchicken) [reply src/main/java/org/kitteh/irc/client/library/feature/twitch/messagetag/Login.java]: I ran Maven install and it was not part of the build. Let me check for a license target.

**Comment by boostchicken** [comment]
@mbax any progress?

**Comment by mbax** [comment]
Sorry, kinda fell off the table. :D

**Comment by mbax** [comment]
I wonder if this button will work nicely...

**Comment by mbax** [comment]
Pulled into `next` branch.
